src/dotnet/CarbonAware.LocationSources.Azure/src/AzureLocationSource.cs

```diff
@@ -19,7 +18,9 @@ public class AzureLocationSource : ILocationSource
 
     private readonly ILogger<AzureLocationSource> _logger;
 
-    private Dictionary<string, AzureRegion> azureRegions;
+    private IDictionary<string, NamedGeoposition> namedGeopositions = new Dictionary<string, NamedGeoposition>();
```

Comment:
I think we still want to have an object that represents the Azure-specific region information, no? Would we not have a scenario where we may need to reference a particular field within the azure data, like an id or something that is unique to Azure that could help in conversions (when it may not just be strict region name to lat/long)

Reply:
We can still add an 'id' or any other field in the NamedGeoposition VO (May have to change it to a better name then).
But, currently the Location object also doesnt take in an 'id' or fields other than lat/long. So, I would keep the scope limited to lat/long for now and create a story to make the model more extensible later.

Reply:
I guess my point is that it seems like we may eventually want an `Azure-specific` object that we can access. So we wouldn't want to change `NamedGeoposition` because that a general `CarbonAware` class but instead, have some model that is Azure specific so that we still have access to that data after the cost of parsing it. But, that's just my opinion and we can just leave a comment or something and come back to it later if it becomes useful

Reply:
resolved during sync